webapp/packages/core-connections/src/ConnectionInfoOriginResource.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { runInAction, toJS } from 'mobx';
+
+import { injectable } from '@cloudbeaver/core-di';
+import { ProjectsService } from '@cloudbeaver/core-projects';
+import { CachedMapResource, isResourceAlias, ResourceKey, resourceKeyList, ResourceKeyUtils } from '@cloudbeaver/core-resource';
+import { DatabaseConnectionOriginFragment, GraphQLService } from '@cloudbeaver/core-sdk';
+import { schemaValidationError } from '@cloudbeaver/core-utils';
+
+import { CONNECTION_INFO_PARAM_SCHEMA, IConnectionInfoParams } from './CONNECTION_INFO_PARAM_SCHEMA';
+import {
+  ConnectionInfoActiveProjectKey,
+  ConnectionInfoProjectKey,
+  ConnectionInfoResource,
+  createConnectionParam,
+  isConnectionInfoParamEqual,
+} from './ConnectionInfoResource';
+
+@injectable()
+export class ConnectionInfoOriginResource extends CachedMapResource<IConnectionInfoParams, DatabaseConnectionOriginFragment> {
```

Comment:
please use ObjectOriginInfoFragment and reexprot it something like `export type ConnectionInfoOrigin = ...`

Reply:
or in case we reuse this type for different entities, `export type EntityOrigin`

Reply:
Unfortunately, we cannot use `ObjectOriginInfoFragment` because it does not include important info about `connectionId` and `projectId.` In a few places, we load this origin info in huge tables, where we load it with the key `CachedMapAllKey` and the `connections` list. We need those ids to match together `connection` and `connectionOriginInfo.`

Reply:
i see, okay